external/vcm/vcm/projection.py

```diff
@@ -0,0 +1,143 @@
+import xarray as xr
+import numpy as np
+
+
+def _center_to_interface(f: np.ndarray) -> np.ndarray:
+    """Interpolate vertically cell centered data to the interface
+    with linearly extrapolated inputs"""
+    f_low = 2 * f[..., 0] - f[..., 1]
+    f_high = 2 * f[..., -1] - f[..., -2]
+    pad = np.concatenate([f_low[..., np.newaxis], f, f_high[..., np.newaxis]], axis=-1)
+    return (pad[..., :-1] + pad[..., 1:]) / 2
+
+
+def _convergence(eddy: np.ndarray, delp: np.ndarray) -> np.ndarray:
+    padded = _center_to_interface(eddy)
+    # pad interfaces assuming eddy = 0 at edges
+    return -np.diff(padded, axis=-1) / delp
+
+
+def cell_center_convergence(
+    eddy: xr.DataArray, delp: xr.DataArray, dim: str = "p"
+) -> xr.DataArray:
+    """Compute vertical convergence of a cell-centered flux.
+
+    This flux is assumed to vanish at the vertical boundaries
+    """
+    return xr.apply_ufunc(
+        _convergence,
+        eddy,
+        delp,
+        input_core_dims=[[dim], [dim]],
+        output_core_dims=[[dim]],
+        dask="parallelized",
+        output_dtypes=[eddy.dtype],
+    )
+
+
+def project_vertical_flux(
```

Comment:
`flux` is the direct object of this name...implying that this function should take `flux` as an argument...but it actually takes a source. To me... "project" also implies that the output has the same type as the input...e.g. P(source) is a also source so that things like `source - P(source) = error`  can be done.

I'm having trouble thinking of an pithy name. `fit_approximate_flux`?

Reply:
I'll use `fit_field_as_flux`. Fits are generally understood to be approximate, and `fit_approximate_flux` still has `flux` as its direct object. I do think it's important to be specifying "vertical" here, but none of the other functions use that naming convention at present.